src/mds/Locker.cc

```diff
@@ -3725,26 +3720,35 @@ void Locker::_update_cap_fields(CInode *in, int dirty, const cref_t<MClientCaps>
 	      << " -> " << m->head.uid
 	      << " for " << *in << dendl;
       pi->uid = m->head.uid;
+      changed = true;
     }
     if (m->head.gid != pi->gid) {
       dout(7) << "  gid " << pi->gid
 	      << " -> " << m->head.gid
 	      << " for " << *in << dendl;
       pi->gid = m->head.gid;
+      changed = true;
     }
     if (m->head.mode != pi->mode) {
       dout(7) << "  mode " << oct << pi->mode
 	      << " -> " << m->head.mode << dec
 	      << " for " << *in << dendl;
       pi->mode = m->head.mode;
+      changed = true;
     }
     if ((features & CEPH_FEATURE_FS_BTIME) && m->get_btime() != pi->btime) {
       dout(7) << "  btime " << oct << pi->btime
 	      << " -> " << m->get_btime() << dec
 	      << " for " << *in << dendl;
       pi->btime = m->get_btime();
+      changed = true;
     }
   }
+
+  if (changed) {
+    pi->ctime = std::max(pi->ctime, ceph_clock_now());
```

Comment:
For the `ctime` change here seems buggy. The client could cache the `ctime` changing such as if the `Fx` or `Ax` is issued. What if the cached `ctime` is flushed to MDS 1 minute later ? The `ctime` here will always be increased by 1 miniute ?